### PR TITLE
Chore/reports read replicas filter

### PR DESCRIPTION
**Pull request by joshenlim**
Allow users to view replica specific information for database reports

image

Allow users to view replica specific information for custom reports
- If a replica is selected, charts will be labelled if the information is for replica, or the project
image

Misc changes:
- Refactored custom reports from mobx to react query
- Deprecated project content mobx store
- Added a button to remove a chart

### CODE REVIEW

File: apps/studio/pages/project/[ref]/reports/api-overview.tsx [hunk 1]
```diff
@@ -108,7 +110,12 @@ export const ApiReport: NextPageWithLayout = () => {
 
 // hook to fetch data
 const useApiReport = () => {
+  const { project } = useProjectContext()
   const { ref: projectRef } = useParams()
+  const state = useDatabaseSelectorStateSnapshot()
+
+  // [Joshen] TODO: Once API support is out
+  const showReadReplicasUI = false // project?.is_read_replicas_enabled
```
Comment (alaister) [inline apps/studio/pages/project/[ref]/reports/api-overview.tsx:118]: Isn't `project?.is_read_replicas_enabled` a thing? 🤨

File: apps/studio/hooks/analytics/useLogsQuery.tsx [hunk 2]
```diff
@@ -27,6 +31,10 @@ const useLogsQuery = (
   projectRef: string,
   initialParams: Partial<LogsEndpointParams> = {}
 ): LogsQueryHook => {
+  // [Joshen] TODO once API is unblocked
```
Comment (alaister) [inline apps/studio/hooks/analytics/useLogsQuery.tsx:34]: Sorry what are we TODOing? 😅

File: apps/studio/components/interfaces/Reports/Reports.tsx [hunk 3]
```diff
@@ -278,59 +192,56 @@ const Reports = () => {
...
+        <div>
+          <h1 className="text-xl text-foreground">{currentReport?.name || 'Reports'}</h1>
+          <p className="text-foreground-light">{currentReport?.description}</p>
+        </div>
+        {hasEdits && (
+          <div className="flex items-center gap-x-2">
+            <Button
+              type="default"
+              onClick={() => setConfig(currentReport?.content)}
+              loading={isSaving}
```
Comment (alaister) [inline apps/studio/components/interfaces/Reports/Reports.tsx:223]:
```suggestion
              disabled={isSaving}
```

File: apps/studio/pages/project/[ref]/reports/api-overview.tsx (hunk 1 above)
Comment (joshenlim) [reply apps/studio/pages/project/[ref]/reports/api-overview.tsx:118]: ^ the API support for read replicas on API reports is not out yet 😬

File: apps/studio/hooks/analytics/useLogsQuery.tsx (hunk 2 above)
Comment (joshenlim) [reply apps/studio/hooks/analytics/useLogsQuery.tsx:34]: sorry! should be more clear in my comments here 😬  i'll update the comment so its clearer!
but all the TODOs here are for the API reports page which BE support is not ready just yet

**Review by alaister — APPROVED** [review]
LGTM!

**Pull request merged**